### PR TITLE
Refactor dotcom resolvers

### PR DESCRIPTION
**Pull request by eseliger**
Remove global assignments and align with pattern used in code intel and batches.

### CODE REVIEW

File: client/web/src/enterprise/site-admin/routes.ts [hunk 1]
```diff
@@ -18,6 +19,7 @@ export const enterpriseSiteAdminAreaRoutes: readonly SiteAdminAreaRoute[] = [
             () => import('./dotcom/customers/SiteAdminCustomersPage'),
             'SiteAdminProductCustomersPage'
         ),
+        condition: () => SHOW_BUSINESS_FEATURES,
```
Comment (eseliger) [inline client/web/src/enterprise/site-admin/routes.ts:22]: This condition was missing. The nav entries were hidden, but the pages were still accessible when navigating to the correct URL.

File: client/web/src/site-admin/SiteAdminArea.tsx [hunk 2]
```diff
@@ -100,17 +100,18 @@ const AuthenticatedSiteAdminArea: React.FunctionComponent<SiteAdminAreaProps> =
...
-                                            key="hardcoded-key"
-                                            path={props.match.url + path}
-                                            exact={exact}
-                                            render={routeComponentProps =>
-                                                render({ ...context, ...routeComponentProps })
-                                            }
-                                        />
-                                    )
+                                    ({ render, path, exact, condition = () => true }) =>
+                                        condition(context) && (
```
Comment (eseliger) [inline client/web/src/site-admin/SiteAdminArea.tsx:104]: we never checked the condition here 🤷‍♂️

File: enterprise/cmd/frontend/internal/dotcom/productsubscription/subscriptions_graphql.go [hunk 3]
```diff
@@ -21,23 +21,23 @@ import (
 	"github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend"
 	"github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/graphqlutil"
 	"github.com/sourcegraph/sourcegraph/enterprise/cmd/frontend/internal/dotcom/billing"
+	"github.com/sourcegraph/sourcegraph/enterprise/cmd/frontend/internal/dotcom/stripeutil"
 	db_ "github.com/sourcegraph/sourcegraph/internal/database"
 	"github.com/sourcegraph/sourcegraph/internal/database/dbutil"
 )
 
-func init() {
-	// TODO(efritz) - de-globalize assignments in this function
```
Comment (efritz) [inline enterprise/cmd/frontend/internal/dotcom/productsubscription/subscriptions_graphql.go:29]: ya did it eri(c|k)

**Review by efritz — APPROVED** [review]
Ok this one legit feels like a HUGE win.

File: enterprise/cmd/frontend/main.go [hunk 4]
```diff
@@ -25,7 +26,6 @@ import (
 	"github.com/sourcegraph/sourcegraph/internal/oobmigration"
 
 	_ "github.com/sourcegraph/sourcegraph/enterprise/cmd/frontend/internal/auth"
```
Comment (unknwon) [inline enterprise/cmd/frontend/main.go:28]: This is unrelated to your PR but since you're on this file... this empty line here doesn't make sense.

**Review by unknwon — APPROVED** [review]
Didn't review very closely but saw many TODOs are being resolved 😁 which feels even nicer.

**Pull request merged**